ui/packages/shared/profile/src/ProfileIcicleGraph/index.tsx

```diff
@@ -354,11 +345,68 @@ const ProfileIcicleGraph = function ProfileIcicleGraphNonMemo({
     return <ErrorContent errorMessage={capitalizeOnlyFirstLetter(error.message)} />;
   }
 
-  if (graph === undefined && arrow === undefined)
-    return <div className="mx-auto text-center">No data...</div>;
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const icicleGraph = useMemo(() => {
+    if (isLoading) {
```

Comment:
we need to also block on `mappingsLoading` here, no?

Reply:
not really, `loading` in the `useEffect` a couple of lines above this code block is already a combination of `flamegraphLoading && profilemetadataLoading` from the parent component's parent component.

```
useEffect(() => {
    if (!loading && (arrow !== undefined || graph !== undefined)) {
      setIsLoading(false);
    } else {
      setIsLoading(true);
    }
  }, [loading, arrow, graph]);
```